### PR TITLE
ScannerCommand: Add an option to skip excluded packages

### PR DESCRIPTION
**Pull request by fviernau**
To be merged after #2270.

### CODE REVIEW

File: model/src/main/kotlin/OrtResult.kt [hunk 1]
```diff
@@ -412,13 +412,19 @@ data class OrtResult(
...
     @JsonIgnore
-    fun getPackages(): Set<CuratedPackage> = analyzer?.result?.packages.orEmpty()
+    fun getPackages(omitExcluded: Boolean = false): Set<CuratedPackage> =
+        analyzer?.result?.packages.orEmpty().filterTo(mutableSetOf()) { pkg ->
+            !omitExcluded || !isExcluded(pkg.pkg.id)
+        }
 
     /**
-     * Return all [Project]s contained in this [OrtResult].
+     * Return all [Project]s contained in this [OrtResult] or only the non-excluded ones if [omitExcluded] is true.
```
Comment (sschuberth) [inline model/src/main/kotlin/OrtResult.kt]: While at it, can we please get docs for `getPackages()`, too?

File: model/src/main/kotlin/OrtResult.kt [hunk 2]
```diff
@@ -412,13 +412,19 @@ data class OrtResult(
     fun getPackage(id: Identifier): CuratedPackage? = packages[id]?.curatedPackage
 
     @JsonIgnore
-    fun getPackages(): Set<CuratedPackage> = analyzer?.result?.packages.orEmpty()
+    fun getPackages(omitExcluded: Boolean = false): Set<CuratedPackage> =
```
Comment (sschuberth) [inline model/src/main/kotlin/OrtResult.kt]: Commit message: Let's not make a mystery about *what* param gets added and change the title to

    OrtResult: Add a param to getProjects() and getPackages() to omit excludes

File: cli/src/main/kotlin/commands/ScannerCommand.kt [hunk 3]
```diff
@@ -65,6 +66,11 @@ class ScannerCommand : CliktCommand(name = "scan", help = "Run existing copyrigh
         ).file(mustExist = true, canBeFile = true, canBeDir = true, mustBeWritable = false, mustBeReadable = true)
     ).single().required()
 
+    private val skipExcluded by option(
+        "--skip-excluded",
+        help = "Do not scan excluded projects or packages. Works only with the '--ort-file' parameter."
+    ).flag(default = false)
```
Comment (sschuberth) [inline cli/src/main/kotlin/commands/ScannerCommand.kt]: I'd prefer to omit the default value here as we do in other places for `flag()`, as it's fair to assume that just declaring a flag should not enable it.

File: cli/src/main/kotlin/commands/ScannerCommand.kt [hunk 4]
```diff
@@ -166,7 +172,8 @@ class ScannerCommand : CliktCommand(name = "scan", help = "Run existing copyrigh
             scanner.scanOrtResult(
                 ortResultFile = ortFile,
                 outputDirectory = absoluteNativeOutputDir,
-                downloadDirectory = absoluteDownloadDir ?: absoluteOutputDir.resolve("downloads")
+                downloadDirectory = absoluteDownloadDir ?: absoluteOutputDir.resolve("downloads"),
+                ignoreExcluded = skipExcluded
```
Comment (sschuberth) [inline cli/src/main/kotlin/commands/ScannerCommand.kt]: The function parameter is also called `skipExcluded`.

File: model/src/main/kotlin/OrtResult.kt (hunk 2 above)
Comment (fviernau) [reply model/src/main/kotlin/OrtResult.kt]: I remember you asked me to pay more attention to the number of chars limit for the commit message title. Thus you have two requests to me which are conflicting, can you please clarify?

Comment (sschuberth) [reply model/src/main/kotlin/OrtResult.kt]: There's a hard limit of 75 chars (to make `git log` read nice in a classic terminal), but this title has only 74 chars. GitHub promotes its own limit of 50 chars for the title, but I regard that as a soft limit (i.e. recommendation) as 50 chars really is quite short, and to me not GitHub but Git itself is the definite source for Git best practices.

File: model/src/main/kotlin/OrtResult.kt (hunk 1 above)
Comment (fviernau) [reply model/src/main/kotlin/OrtResult.kt]: Done.

File: model/src/main/kotlin/OrtResult.kt (hunk 2 above)
Comment (fviernau) [reply model/src/main/kotlin/OrtResult.kt]: Done.

File: cli/src/main/kotlin/commands/ScannerCommand.kt (hunk 3 above)
Comment (fviernau) [reply cli/src/main/kotlin/commands/ScannerCommand.kt]: Done.

File: cli/src/main/kotlin/commands/ScannerCommand.kt (hunk 4 above)
Comment (fviernau) [reply cli/src/main/kotlin/commands/ScannerCommand.kt]: Done.

**Comment by sschuberth** [comment]
I'll continue the review once #2270 has been merged and this has been rebased.

**Pull request merged**